### ISSUE TITLE
SQLite generation fails due to implicit foreign key column

### ISSUE THREAD

**Issue by hmnd** [stephenafamo/bob#167]
I was trying to set up bob in my existing project, when I ran into the following cryptic error:

`unable to fetch table data: sql: Scan error on column index 4, name "to": converting NULL to string is unsupported`

I found that this error was coming from the foreign key list unmarshaling [here](https://github.com/stephenafamo/bob/blob/891dd5a56ba2731ab942801e1f95b5ae544f214f/gen/bobgen-sqlite/driver/sqlite.go#L374). It was caused by my tables having implicitly defined columns on my foreign keys; something like this:

```sql
CREATE TABLE instance(
  id INTEGER PRIMARY KEY
);
CREATE TABLE endpoint(
  id INTEGER PRIMARY KEY,
  instance_id REFERENCES instance
);
```

I know this isn't best practice to begin with and have since switched my schema to explicitly define the column, but perhaps a clearer error might help other users who run into a similar issue.

Thanks for this cool library!

**Comment by stephenafamo** [comment]
Thanks for finding this bug.

I did not account for implicit foreign keys when creating the SQLite plugin so I would look into that and fix. I checked and this syntax is also valid for Postgres, so I would have to check that too.

Whether or not it is *best practice*, it is perfectly valid and not deprecated, so **Bob** should definitely support it.

**Comment by stephenafamo** [comment]
I finally got some time to work on this. It should be fixed in #208 

Postgres already reports the correct columns so there was no need to change anything in the psql driver.
